lib/matplotlib/colors.py

```diff
@@ -484,6 +484,21 @@ def makeMappingArray(N, data, gamma=1.0):
     return _create_lookup_table(N, data, gamma)
 
 
+def _warn_if_global_cmap_modified(cmap):
+    if getattr(cmap, '_global', False):
+        cbook.warn_deprecated(
+            "3.3",
+            message="You are modifying the state of a globally registered "
+                    "colormap. In future versions, you will not be able "
+                    "to modify a globally registered colormap directly. "
+                    "To eliminate this warning until then, you can make "
+                    "a copy of the requested colormap before modifying it. ",
+            alternative="To modify a colormap without overwriting the "
+                        "global state, you can make a copy of the colormap "
+                        f"first. cmap = copy.copy(mpl.cm.get_cmap({cmap.name})"
```

Comment:
This is a pretty verbose alternative for something that I expect to catch a fair number of users.  I am in favor of adding the `copy=True` kwarg to `get_cmap` in the same move as we add this warning so we can provide a less verbose (if less back-compatible) option to users.

Reply:
I decided to just change the wording here for now, and put `.copy()` instead of `copy.copy()`. I think the wording was just a little long, so I tidied it up. The alternative is actually less verbose than `copy=True`. I also don't really know if we should add a keyword for one version that will just be removed a version later when it is unnecessary. Let me know what you think...